### DISCUSSION TITLE
Placement of the ToolTip() call

### DISCUSSION THREAD

**Discussion by kardoka** [iTwin/itwinjs-core#2351] [Q&A]
NOTE: As a mistake, I've posted this as a issue instead of starting a Q&A discussion thread.

Greetings,

I'm getting a hook call error while creating a new Tooltip object.

MyTooltipProps class:
```
class MyTooltipProps implements TooltipProps {
  children: React.ReactNode;
  placement: TooltipPlacement;
  target: HTMLElement;
  visible: boolean

  constructor(children: React.ReactNode, placement: TooltipPlacement, target: HTMLElement, visible: boolean) {
    this.children = children;
    this.placement = placement;
    this.target = target;
    this.visible = visible;
  }
}
```

The method for creating tooltips:
```
const createTooltips = (targetElement: HTMLElement) => {
    const children = <h1>Hello.</h1>;
    const placement = "top";
    const target = targetElement;
    const visible = true;

    const toolTipProps: MyTooltipProps = new MyTooltipProps(children, placement, target, visible);
    const toolTip = Tooltip(toolTipProps);
    return toolTip;
  }
```

...which is called in `App.tsx`:
```
  const onIModelConnected = (_imodel: IModelConnection) => {
    let element: HTMLElement = document.createElement('div');
    element = document.getElementById('uifw-contentlayout-div') as HTMLElement;
    createTooltips(element);
    IModelApp.viewManager.onViewOpen.addOnce(async (vp: ScreenViewport) => {
      IModelApp.viewManager.addDecorator(new SmartDeviceDecorator(vp));
    })
  }
```

Could the issue be with creating the tooltip at the wrong place in the code? What would be a correct example use-case of the Tooltip method?

**Comment by aruniverse** [comment]
Have you taken a look at the following [ToolTip Sample](
https://www.itwinjs.org/sample-showcase/?group=Viewer+Features&sample=tooltip-customize-sample&gallery=false&imodel=Metrostation+Sample) and how to [go from the sample to your own viewer](https://developer.bentley.com/tutorials/adding-showcase-widgets-to-your-itwin-viewer/) ?

**Reply by kardoka** [reply]
Thank you, I am aware of that sample, but tooltips in the sample are created without creating a Tooltip object. So, the only way to add tooltips to the UI is by extending ToolAdmin? If so, then where can I use the Tooltip class?
